src/main/java/net/sf/jabref/logic/util/strings/StringUtil.java

```diff
@@ -646,7 +646,14 @@ public static String expandAuthorInitials(String name) {
                 }
                 for (int j = 1; j < names.length; j++) {
                     if (j == 1) {
-                        sb.append(Author.addDotIfAbbreviation(names[j]));
+                        String[] splittedNames = names[j].split(" ");
+                        if (splittedNames.length > 0) {
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[0]));
+                        }
+                        for (int k = 1; k < splittedNames.length; k++) {
+                            sb.append(' ');
+                            sb.append(Author.addDotIfAbbreviation(splittedNames[k]));
+                        }
```

Comment:
Ah, I see. According to [the JAvaDoc of StringJoiner](https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html), one can also transform the inputs (achieved by `map` in my proposal). Since the iteration on the split result (an array) is easy via an foreach, this should easily work without some if checking for non-emptyness (`splittedNames.length > 0`) etc.